docs/using-hive.md

```diff
@@ -2,162 +2,326 @@
+Alternatively, you can specify an individual OpenShift release image in the `ClusterDeployment` `spec.provisioning.releaseImage` property.
 
-An example ClusterImageSet:
+An example `ClusterImageSet`:
 
 ```yaml
 apiVersion: hive.openshift.io/v1
 kind: ClusterImageSet
 metadata:
-  name: openshift-v4.2.0
+  name: openshift-v4.3.0
 spec:
-  releaseImage: quay.io/openshift-release-dev/ocp-release:4.2.0
+  releaseImage: quay.io/openshift-release-dev/ocp-release:4.3.0
+```
+
+### Cloud credentials
+
+Hive requires credentials to the cloud account into which it will install OpenShift clusters.
+
+#### AWS
+
+Create a `secret` containing your AWS access key and secret access key:
+
+```yaml
+apiVersion: v1
+data:
+  aws_access_key_id: REDACTED
+  aws_secret_access_key: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-aws-creds
+  namespace: mynamespace
+type: Opaque
 ```
 
-Alternatively you can specify release image overrides directly on your ClusterDeployment when you create one.
+#### Azure
 
-Cluster provisioning begins when a caller creates a ClusterDeployment CR, which is the core Hive resource to control the lifecycle of a cluster.
+Create a `secret` containing your Azure service principal:
 
-An example ClusterDeployment:
+```yaml
+apiVersion: v1
+data:
+  osServicePrincipal.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-azure-creds
+  namespace: mynamespace
+type: Opaque
+```
+
+#### GCP
+
+Create a `secret` containing your GCP service account key:
+
+```yaml
+apiVersion: v1
+data:
+  osServiceAccount.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-gcp-creds
+  namespace: mynamespace
+type: Opaque
+```
+
+### SSH Key Pair
+
+(Optional) Hive uses the provided ssh key pair to ssh into the machines in the remote cluster. Hive connects via ssh to gather logs in the event of an installation failure. The ssh key pair is optional, but neither the user nor Hive will be able to ssh into the machines if it is not supplied.
+
+Create a Kubernetes secret containing a ssh key pair (typically generated with `ssh-keygen`)
+
+```yaml
+apiVersion: v1
+data:
+  ssh-privatekey: REDACTED
+  ssh-publickey: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-ssh-key
+  namespace: mynamespace
+type: Opaque
+```
+
+### InstallConfig
+
+The OpenShift installer `InstallConfig` must be stored in a `secret` and referenced in the `ClusterDeployment`. This allows Hive to more easily support installing multiple versions of OpenShift.
+
+Example `install-config.yaml` for AWS:
+
+```yaml
+apiVersion: v1
+baseDomain: hive.example.com
+compute:
+- name: worker
+  platform:
+    aws:
+      rootVolume:
+        iops: 100
+        size: 22
+        type: gp2
+      type: m4.xlarge
+  replicas: 3
+controlPlane:
+  name: master
+  platform:
+    aws:
+      rootVolume:
+        iops: 100
+        size: 22
+        type: gp2
+      type: m4.xlarge
+replicas: 3
```

Comment:
1) The `replicas` field is ignored for control plane.
2) If you still want to include the `replicas` field, then it needs to be indented one level so that it falls under the `controlPlane` field.